.github/workflows/maven.yml

```diff
@@ -39,12 +39,12 @@ jobs:
           mvn -version
 
       - name: Build and test (headless with xvfb in Linux)
-        run: xvfb-run mvn clean verify -PuseJenkinsSnapshots
+        run: xvfb-run mvn clean verify -PuseJenkinsSnapshots -Dsurefire.rerunFailingTestsCount=3
         if: matrix.os == 'ubuntu-latest'
         working-directory: org.eclipse.xtext.full.releng
 
       - name: Build and test (in other OSes)
-        run: mvn clean verify -PuseJenkinsSnapshots
+        run: mvn clean verify -PuseJenkinsSnapshots -Dsurefire.rerunFailingTestsCount=3
```

Comment:
Line 77 does not need this?

Reply:
@szarnekow No because that builds only Maven artifacts and their tests are not flaky; they don't use Eclipse OSGI to run.